source/Mlos.Model.Services.Client/OptimizationProblemExtension.cs

```diff
@@ -22,23 +22,8 @@ internal static class OptimizationProblemExtension
         /// <returns></returns>
         internal static OptimizerService.OptimizationProblem ToOptimizerServiceOptimizationProblem(this OptimizationProblem optimizationProblem)
         {
-            var optimizerOptimizationProblem = new OptimizerService.OptimizationProblem
-            {
-                ParameterSpace = new OptimizerService.Hypergrid
-                {
-                    HypergridJsonString = optimizationProblem.ParameterSpace.ToJson(),
-                },
-
-                ContextSpace = new OptimizerService.Hypergrid
-                {
-                    HypergridJsonString = optimizationProblem.ContextSpace != null ? optimizationProblem.ContextSpace.ToJson() : string.Empty,
-                },
-
-                ObjectiveSpace = new OptimizerService.Hypergrid
-                {
-                    HypergridJsonString = optimizationProblem.ObjectiveSpace.ToJson(),
-                },
-            };
+            var encoder = new OptimizerServiceEncoder();
+            var optimizerOptimizationProblem = encoder.EncodeOptimizationProblem(optimizationProblem);

```

Comment:
use full type 
(don't use var unless we know the type)

Reply:
[zaporter/grpc 4a74b7b5] Refactored to use C# syntax tricks